pkg/frontend/querymiddleware/shard_active_series.go

```diff
@@ -330,10 +359,15 @@ func (s *shardActiveSeriesMiddleware) writeMergedResponse(ctx context.Context, c
 	} else {
 		span.LogFields(otlog.String("encoding", "none"))
 	}
+	buf := s.getBufferFromPool()
+	defer s.putBuffer(buf)
+
+	stream := jsoniter.ConfigFastest.BorrowStream(out)
+	stream.SetBuffer(buf[:0])
```

Comment:
```suggestion
	stream := jsoniter.ConfigFastest.BorrowStream(out)
```

From what I can tell in my testing, setting the buffer here is either no better or sometimes slightly worse than letting the existing buffer in the stream get re-used. It gets reset here: https://github.com/grafana/mimir/blob/91806863d75be5865404b0016a2a25aec5a0b496/vendor/github.com/json-iterator/go/stream.go#L40

WDYT?

```
goos: darwin
goarch: arm64
pkg: github.com/grafana/mimir/pkg/frontend/querymiddleware
                                                          │      old      │                 new                  │
                                                          │    sec/op     │    sec/op     vs base                │
ActiveSeriesMiddlewareMergeResponses/num-responses-2-10      5.603µ ±  7%   5.543µ ±  4%        ~ (p=0.289 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-4-10      6.975µ ±  8%   6.857µ ±  4%        ~ (p=0.393 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-8-10     10.255µ ± 19%   9.546µ ±  2%   -6.91% (p=0.035 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-16-10     14.29µ ± 11%   14.07µ ±  3%        ~ (p=0.280 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-32-10     25.05µ ±  9%   24.40µ ± 12%        ~ (p=0.853 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-64-10     52.65µ ±  8%   51.51µ ±  1%   -2.15% (p=0.009 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-128-10    111.6µ ±  8%   104.1µ ±  3%   -6.73% (p=0.007 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-256-10    241.7µ ±  3%   217.3µ ±  8%  -10.06% (p=0.000 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-512-10    495.5µ ±  4%   472.1µ ±  4%   -4.72% (p=0.035 n=10)
geomean                                                      35.98µ         34.46µ         -4.21%

                                                          │     old      │                 new                  │
                                                          │     B/op     │     B/op      vs base                │
ActiveSeriesMiddlewareMergeResponses/num-responses-2-10     1.571Ki ± 0%   1.571Ki ± 0%        ~ (p=0.628 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-4-10     1.821Ki ± 0%   1.821Ki ± 0%        ~ (p=0.087 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-8-10     2.322Ki ± 0%   2.322Ki ± 0%        ~ (p=0.263 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-16-10    3.323Ki ± 0%   3.323Ki ± 0%        ~ (p=0.902 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-32-10    5.326Ki ± 0%   5.327Ki ± 0%        ~ (p=0.835 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-64-10    9.333Ki ± 0%   9.337Ki ± 0%   +0.04% (p=0.039 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-128-10   17.35Ki ± 0%   17.36Ki ± 0%   +0.03% (p=0.015 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-256-10   33.42Ki ± 0%   33.44Ki ± 0%   +0.04% (p=0.011 n=10)
ActiveSeriesMiddlewareMergeResponses/num-responses-512-10   86.89Ki ± 0%   65.96Ki ± 0%  -24.08% (p=0.000 n=10)
geomean                                                     7.250Ki        7.032Ki        -3.00%

                                                          │     old     │                 new                  │
                                                          │  allocs/op  │  allocs/op   vs base                 │
ActiveSeriesMiddlewareMergeResponses/num-responses-2-10      28.00 ± 0%    28.00 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-4-10      38.00 ± 0%    38.00 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-8-10      58.00 ± 0%    58.00 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-16-10     98.00 ± 0%    98.00 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-32-10     178.0 ± 0%    178.0 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-64-10     338.0 ± 0%    338.0 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-128-10    658.0 ± 0%    658.0 ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-256-10   1.298k ± 0%   1.298k ± 0%       ~ (p=1.000 n=10) ¹
ActiveSeriesMiddlewareMergeResponses/num-responses-512-10   2.579k ± 0%   2.578k ± 0%  -0.04% (p=0.000 n=10)
geomean                                                      210.2         210.2       -0.00%
¹ all samples are equal

```

Reply:
Sounds good! 👍 

_changed @ cdee5f64192053d1938d8b4f8c351b5f8c22081b_